$NetBSD$

* use find_package to detect Snappy to set any build flags well

--- CMakeLists.txt.orig	2019-05-03 20:24:26.000000000 +0000
+++ CMakeLists.txt
@@ -34,9 +34,13 @@ test_big_endian(LEVELDB_IS_BIG_ENDIAN)
 include(CheckIncludeFile)
 check_include_file("unistd.h" HAVE_UNISTD_H)
 
+find_package(Snappy)
+if(Snappy_FOUND)
+  set(HAVE_SNAPPY 1)
+endif()
+
 include(CheckLibraryExists)
 check_library_exists(crc32c crc32c_value "" HAVE_CRC32C)
-check_library_exists(snappy snappy_compress "" HAVE_SNAPPY)
 check_library_exists(tcmalloc malloc "" HAVE_TCMALLOC)
 
 include(CheckCXXSymbolExists)
@@ -248,9 +252,9 @@ endif(HAVE_CLANG_THREAD_SAFETY)
 if(HAVE_CRC32C)
   target_link_libraries(leveldb crc32c)
 endif(HAVE_CRC32C)
-if(HAVE_SNAPPY)
-  target_link_libraries(leveldb snappy)
-endif(HAVE_SNAPPY)
+if(Snappy_FOUND)
+  target_link_libraries(leveldb Snappy::snappy)
+endif(Snappy_FOUND)
 if(HAVE_TCMALLOC)
   target_link_libraries(leveldb tcmalloc)
 endif(HAVE_TCMALLOC)
